File: tests/fixtures/cpp_basic/tests/payment_service_test.cpp
#include "acme/payment_service.hpp"

namespace acme {

void authorize_allows_valid_payment() {
  PaymentGateway gateway;
  PaymentService service(&gateway);
  service.authorize(PaymentRequest{42});
}

void authorize_fails_invalid_payment() {
  PaymentGateway gateway;
  PaymentService service(&gateway);
  service.authorize(PaymentRequest{0});
}

}  // namespace acme

File: tests/fixtures/java_basic/src/test/java/acme/PaymentServiceTest.java
package acme;

class PaymentServiceTest {
  void authorize_allows_valid_payment() {
    PaymentService service = new PaymentService(new PaymentGateway());
    service.authorize(new PaymentRequest("u1", 42));
  }

  void authorize_fails_invalid_payment() {
    PaymentService service = new PaymentService(new PaymentGateway());
    service.authorize(null);
  }
}
